packages/admin/cms-admin/src/dependencies/DependencyContext.tsx

```diff
@@ -0,0 +1,35 @@
+import { ApolloClient } from "@apollo/client";
+import * as React from "react";
+
+import { GQLDependency } from "../graphql.generated";
+
+export interface DependencyRenderInfoOptions {
+    data: Pick<GQLDependency, "rootGraphqlObjectType" | "rootId" | "rootColumnName" | "jsonPath">;
+    apolloClient: ApolloClient<unknown>;
+    contentScopeUrl: string;
+}
+export interface DependencyRenderInfo {
+    type: React.ReactNode;
+    name: React.ReactNode;
+    secondaryInfo?: React.ReactNode;
+    url?: string;
+}
+
+export type GetRenderInfo = (id: string, options: DependencyRenderInfoOptions) => Promise<DependencyRenderInfo> | DependencyRenderInfo;
+
+export interface DependencyConfig {
+    dependencyRenderInfoProvider?: {
+        [key: string]: GetRenderInfo;
+    };
+}
+
+const DependencyContext = React.createContext<DependencyConfig | undefined>(undefined);
+
+export const DependencyProvider: React.FunctionComponent<{ value: DependencyConfig }> = ({ children, value }) => {
```

Comment:
suggestion how this could be changed to:

```
                        <DependenciesConfig
                            rootEntities={{
                                Page: {
                                    dependencyComponent: PageDependency
                                }
                            }}
                        >
```
- better name; DependencyProvider doesn't provide dependencies
- not using value prop
- change the renderInfo api to a standard component, that could look like:

```
function PageDependency(props: { id: string, rootGraphqlObjectType: string, rootId: string, rootColumnName: string, jsonPath: string }) => {
    const { data } = useQuery<GQLPageDependencyQuery, GQLPageDependencyQueryVariables>(Page.dependencyQuery, variables: { id: props.id });

    return <Dependency type={Page.displayName} name={Page.getName(props)}
        secondaryInfo={Page.getSecondaryInformation?.(props)} url={Page.getUrl(data, {
            rootColumn: dependencyData.rootColumnName,
            jsonPath: dependencyData.jsonPath,
            contentScopeUrl: contentScopeUrl,
        })}>
};
```

If moving those parts into Page (getNAme, getUrl etc is really a good idea I'm not sure.

- That dependency stuff is not at one place (DependencyProvider(or DependenciesConfig) + Page instead of just DependencyProvider)
- it is only used by application code, so there is no benefit from a generic interface (or is the plan to change that?)

Reply:
> * change the renderInfo api to a standard component, that could look like:
> 
> ```
> function PageDependency(props: { id: string, rootGraphqlObjectType: string, rootId: string, rootColumnName: string, jsonPath: string }) => {
>     const { data } = useQuery<GQLPageDependencyQuery, GQLPageDependencyQueryVariables>(Page.dependencyQuery, variables: { id: props.id });
> 
>     return <Dependency type={Page.displayName} name={Page.getName(props)}
>         secondaryInfo={Page.getSecondaryInformation?.(props)} url={Page.getUrl(data, {
>             rootColumn: dependencyData.rootColumnName,
>             jsonPath: dependencyData.jsonPath,
>             contentScopeUrl: contentScopeUrl,
>         })}>
> };
> ```

I thought about something like this, providing a component instead of the render information. I figured that it would reduce flexibility because there is only one way the information can be displayed (via this component). 
However, I'm unsure if we ever actually need this flexibility. Probably, the information will always be displayed in a list (like I do it now in the DAM). At least I can't really think of another case right now